tensornetwork/tests/network_test.py

```diff
@@ -490,18 +490,28 @@ def test_contract_between(backend):
   tn.connect(a[0], b[3])
   tn.connect(b[1], a[3])
   tn.connect(a[1], b[0])
-  edge_a = a[2]
-  edge_b = b[2]
-  c = tn.contract_between(a, b, name="New Node")
-  c.reorder_edges([edge_a, edge_b])
+  output_axis_names = ["a2", "b2"]
+  c = tn.contract_between(a, b, name="New Node", axis_names=output_axis_names)
   tn.check_correct({c})
   # Check expected values.
   a_flat = np.reshape(np.transpose(a_val, (2, 1, 0, 3)), (4, 30))
   b_flat = np.reshape(np.transpose(b_val, (2, 0, 3, 1)), (4, 30))
   final_val = np.matmul(a_flat, b_flat.T)
   assert c.name == "New Node"
+  assert c.axis_names == output_axis_names
   np.testing.assert_allclose(c.tensor, final_val)
 
+  # Test output_edge_order
+  tn.connect(a[0], b[3])
+  tn.connect(b[1], a[3])
+  tn.connect(a[1], b[0])
+  output_axis_names = ["b2", "a2"]
+  c = tn.contract_between(a, b, name="New Node", axis_names=output_axis_names,
+                          output_edge_order=[a[2], b[2]])
+  assert c.name == "New Node"
+  assert c.axis_names == output_axis_names
+  np.testing.assert_allclose(c.tensor, final_val.T)
+
```

Comment:
Can you break this out to a new test? Reusing already contracted nodes is kinda dangerous, and we may change that behavior in the future.